frameworks/Elixir/phoenix/web/controllers/page_controller.ex

```diff
@@ -60,16 +60,17 @@ defmodule Hello.PageController do
       ArgumentError -> 1
     end
 
-    conn
-    |> put_resp_content_type("application/json", nil)
-    |> send_resp(200, Poison.encode!(Enum.map(1..q, fn _ ->
+    data = for _ <- 1..q do
       id = :rand.uniform(10000)
       num = :rand.uniform(10000)
       w = Repo.get(World, id)
-      changeset = World.changeset(w, %{randomnumber: num})
-      Repo.update(changeset)
-      %{id: id, randomnumber: num}
-    end)))
+      changeset = Ecto.Changeset.change(w, randomnumber: num)
+      Repo.update!(changeset)
```

Comment:
Can't this be optimized by a single update query?

```
from w in World, where: w.id == ^id,
update: [set: [randomnumber: ^num]]
```

Reply:
this is benchmark requirement

Reply:
It's very likely there are more optimisation opportunities, but those have to be explored with some more careful profiling.